src/formatDate.js

```diff
@@ -50,7 +9,64 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatedDate = [];
+  const sortedDate = {};
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateArray = date.split(oldSeparator);
+
+  const days = 'DD';
+  const months = 'MM';
+  const shortYears = 'YY';
+  const fullYears = 'YYYY';
+
+  for (let i = 0; i < dateArray.length; i++) {
+    sortedDate[fromFormat[i]] = dateArray[i];
+  }
+
+  for (let i = 0; i < dateArray.length; i++) {
+    switch (toFormat[i]) {
+      case days:
+        formatedDate.push(sortedDate[days]);
+        break;
+
+      case months:
+        formatedDate.push(sortedDate[months]);
+        break;
+
+      case fullYears:
+        let lengthenedYears;
+
+        if (sortedDate[shortYears] < 30) {
```

Comment:
This part looks like a nested if statement (case + if) 
![image](https://user-images.githubusercontent.com/57747481/205730492-eef8397d-b998-476f-9dd4-b04590926624.png)